plugins/inputs/cpufreq/cpufreq.go

```diff
@@ -0,0 +1,157 @@
+// +build linux
+
+package cpufreq
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const defaultHostSys = "/sys"
+
+type CPUFreq struct {
+	Log             telegraf.Logger `toml:"-"`
+	CPUs            []CPU           `toml:"-"`
+	PathSysfs       string          `toml:"host_sys"`
+	GatherThrottles bool            `toml:"gather_throttles"`
+}
+
+type CPU struct {
+	ID    string
+	Path  string
+	Props map[string]*os.File
+}
+
+var sampleConfig = `
+  ## Path for sysfs filesystem.
+  ## See https://www.kernel.org/doc/Documentation/filesystems/sysfs.txt
+  ## Defaults:
+  # host_sys = "/sys"
+
+  ## Gather CPU throttles per core
+  ## Defaults:
+  # gather_throttles = false
+`
+
+func (g *CPUFreq) SampleConfig() string {
+	return sampleConfig
+}
+
+func (g *CPUFreq) Description() string {
+	return "Collect current CPU's frequencies"
+}
+
+func (g *CPUFreq) Init() error {
+	if os.Getenv("HOST_SYS") != "" {
+		g.PathSysfs = os.Getenv("HOST_SYS")
+	} else {
+		g.PathSysfs = defaultHostSys
+	}
```

Comment:
Because of `if-else`, `host_sys` paramater from config will be never taken into consideration - it will be always overwritten by content of `os.Getenv("HOST_SYS")` or by `defaultHostSys`.

I would wrap this `if-else` with:
```
if g.PathSysfs == "" {
...
}
```